pkg/frontend/admin_hive_syncset_list.go

```diff
@@ -0,0 +1,60 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"reflect"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/ugorji/go/codec"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+func (f *frontend) listAdminHiveSyncSet(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	namespace := r.URL.Query().Get("namespace")
+	label := r.URL.Query().Get("lable")
+	isSyncSet := r.URL.Query().Has("isSyncSet")
+	b, err := f._listAdminHiveSyncSet(ctx, namespace, label, isSyncSet)
+
+	if cloudErr, ok := err.(*api.CloudError); ok {
+		api.WriteCloudError(w, cloudErr)
+		return
+	}
+
+	adminReply(log, w, nil, b, err)
+}
+
+func (f *frontend) _listAdminHiveSyncSet(ctx context.Context, namespace string, label string, isSyncSet bool) ([]byte, error) {
+	// defaults to listing selectorSyncSets for an AKS instance
+	ssType := reflect.TypeOf(hivev1.SelectorSyncSet{})
```

Comment:
Looks like this is repeated in a couple of places, I think we should take this out from these specific functions `_list/get...` and call it outside so we don't duplicate this code. And we should call `_list*SyncSet, _list*SelectorSyncSet` explicitly to set this up in the long-run. For example, modifying code for SyncSet only will not cause more if statements. This also promotes short and simple functions.